pkg/mimirtool/commands/analyse_grafana.go

```diff
@@ -85,7 +82,10 @@ func AnalyzeGrafana(ctx context.Context, c *sdk.Client, folders []string) (*anal
 		if filterOnFolders && !slices.Contains(folders, link.FolderTitle) {
 			continue
 		}
-		data, _, err := c.GetRawDashboardByUID(ctx, link.UID)
+		fetchCtx, cancel := context.WithTimeout(context.Background(), readTimeout)
```

Comment:
I'd make this a child of the context passed to `AnalyzeGrafana`:

```suggestion
		fetchCtx, cancel := context.WithTimeout(ctx, readTimeout)
```

Reply:
addressed.